### PR TITLE
Various cleanups around pre-TyCtxt queries and functions

### PR DESCRIPTION
**Pull request by oli-obk**
part of #105462

based on https://github.com/rust-lang/rust/pull/106776 (everything starting at [0e2b39f](https://github.com/rust-lang/rust/pull/106810/commits/0e2b39fd1ffde51b50d45ccbe41de52b85136b8b) is new in this PR)

r? @petrochenkov 

I think this should be most of the uncontroversial part of #105462.

### CODE REVIEW

**Comment by oli-obk** [comment]
@rust-timer build dc88ba6552d55a6b2fa3a3ea6fbad64e10e134e5

**Comment by petrochenkov** [comment]
Perf is ok, rebase is needed.
@rustbot author

File: compiler/rustc_driver/src/lib.rs [hunk 1]
```diff
@@ -362,16 +362,14 @@ fn run_compiler(
                 return early_exit();
             }
 
-            queries.prepare_outputs()?;
+            queries.global_ctxt()?;
```
[the remark this answers is not in the document]
Comment (petrochenkov) [reply compiler/rustc_driver/src/lib.rs:336]: That said, `--emit dep-info` is supposed to be as cheap as possible.
So after the dust settles the `write_out_deps` logic will still need to be moved to as earlier compilation point as possible.

File: compiler/rustc_session/src/session.rs [hunk 2]
```diff
@@ -147,9 +156,8 @@ pub struct Session {
     pub target_tlib_path: Lrc<SearchPath>,
     pub parse_sess: ParseSess,
     pub sysroot: PathBuf,
-    /// The name of the root source file of the crate, in the local file system.
-    /// `None` means that there is no source file.
-    pub local_crate_source_file: Option<PathBuf>,
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply compiler/rustc_session/src/session.rs:152]: It is only used once per compilation, so caching wouldn't really change anything.

The only usages are

* cranelift debug info
* codegen_llvm debug info
* an error path
* save_analysis
* rustdoc rendering

File: compiler/rustc_driver/src/pretty.rs [hunk 3]
```diff
@@ -484,7 +478,7 @@ pub fn print_after_hir_lowering<'tcx>(
 fn print_with_analysis(
     tcx: TyCtxt<'_>,
     ppm: PpMode,
-    ofile: Option<&Path>,
+    ofile: &Option<PathBuf>,
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply compiler/rustc_driver/src/pretty.rs]: This argument does not exist anymore at the end of the PR

File: compiler/rustc_driver/src/pretty.rs [hunk 4]
```diff
@@ -358,7 +359,7 @@ fn get_source(input: &Input, sess: &Session) -> (String, FileName) {
     (src, src_name)
 }
 
-fn write_or_print(out: &str, ofile: Option<&Path>, sess: &Session) {
+fn write_or_print(out: &str, ofile: &Option<PathBuf>, sess: &Session) {
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply compiler/rustc_driver/src/pretty.rs]: This argument does not exist anymore at the end of this PR

File: compiler/rustc_driver/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (oli-obk) [reply compiler/rustc_driver/src/lib.rs:336]: This shows up very obviously in perf (I have hit this before when I broke something), so we won't accidentally regress anything here.

File: compiler/rustc_session/src/session.rs [hunk 5]
```diff
@@ -1,7 +1,9 @@
 use crate::cgu_reuse_tracker::CguReuseTracker;
 use crate::code_stats::CodeStats;
 pub use crate::code_stats::{DataTypeKind, FieldInfo, SizeKind, VariantInfo};
-use crate::config::{self, CrateType, InstrumentCoverage, OptLevel, OutputType, SwitchWithOptPath};
+use crate::config::{
+    self, CrateType, Input, InstrumentCoverage, OptLevel, OutputType, SwitchWithOptPath,
+};
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply compiler/rustc_session/src/session.rs]: I made a fixup to the commit that introduced this multi-line import and force pushed

File: compiler/rustc_driver/src/lib.rs [hunk 6]
```diff
@@ -26,6 +26,7 @@ use rustc_errors::registry::{InvalidErrorCode, Registry};
 use rustc_errors::{ErrorGuaranteed, PResult};
```
Comment (bjorn3) [inline compiler/rustc_driver/src/lib.rs:26]: commit message nit: s/toghether/together

File: compiler/rustc_session/src/session.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (petrochenkov) [reply compiler/rustc_session/src/session.rs:152]: Ok.

[the remark this answers is not in the document]
Comment (petrochenkov) [reply compiler/rustc_session/src/session.rs:152]: Ok.

**Comment by compiler-errors** [comment]
This is next in the bors queue, but bors isn't starting on it.

@bors r-

**Comment by SmiteWindows** [comment]
@petrochenkov bors ?

**Comment by Mark-Simulacrum** [comment]
@bors r=petrochenkov

Let's see if this helps. Previously PR CI wasn't running either so I wonder if some internal GitHub state was confused.

**Pull request merged**